ir/shared/src/main/scala/org/scalajs/ir/Types.scala

```diff
@@ -278,7 +278,11 @@ object Types {
     case DoubleType  => DoubleLiteral(0.0)
     case StringType  => StringLiteral("")
     case UndefType   => Undefined()
-    case _           => Null()
+
+    case NullType | AnyType | _:ClassType | _:ArrayType | _:RecordType => Null()
```

Comment:
@sjrd is `RecordType` here correct? IIUC `RecordType`s are value types, so needing a zero of them would mean something went horribly wrong, no?